src/main.ts

```diff
@@ -11,15 +11,19 @@ import {
 
 import { Controller } from './controller';
 import { IFileService } from '@core/interfaces/file-service.interface';
-import { PLATFORMS } from '@core/constants/main.constants';
+import { MacFilesService } from './services/mac-files.service';
 
-const isOSWindows = () => process.platform === PLATFORMS.WINDOWS;
+const getOS = () => process.platform;
+
+const OSService = {
+  linux: () => new LinuxFilesService(streamService),
+  win32: () => new WindowsFilesService(streamService),
+  darwin: () => new MacFilesService(streamService),
+};
 
 const streamService: StreamService = new StreamService();
 
-const fileService: IFileService = isOSWindows()
-  ? new WindowsFilesService(streamService)
-  : new LinuxFilesService(streamService);
+const fileService: IFileService = OSService[getOS()]();
```

Comment:
Following above...
```suggestion
const fileService: IFileService = new OSService[getOS()](streamService);
```

Reply:
Done in 4145810
Thank you very much for all your suggestions @tonivj5 :D